From 80dfef14c9f72460bdd5692e275fcbc1dce6d5be Mon Sep 17 00:00:00 2001
From: Michal Schmidt <mschmidt@redhat.com>
Date: Fri, 5 Sep 2014 19:49:39 -0400
Subject: [ethernet] bnx2x: get rid of SET_ETHTOOL_OPS

Message-id: <1409946615-5862-35-git-send-email-mschmidt@redhat.com>
Patchwork-id: 91578
O-Subject: [RHEL7.1 PATCH 34/70] bnx2x: get rid of SET_ETHTOOL_OPS
Bugzilla: 1089402
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1089402

part of upstream commit 7ad24ea4bf620a32631d7b3069c3e30c078b0c3e
("net: get rid of SET_ETHTOOL_OPS")
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_ethtool.c b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_ethtool.c
index b6de05e..0322409 100644
--- a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_ethtool.c
+++ b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_ethtool.c
@@ -3506,8 +3506,6 @@ static const struct ethtool_ops bnx2x_vf_ethtool_ops = {
 
 void bnx2x_set_ethtool_ops(struct bnx2x *bp, struct net_device *netdev)
 {
- if (IS_PF(bp))
-  SET_ETHTOOL_OPS(netdev, &bnx2x_ethtool_ops);
- else /* vf */
-  SET_ETHTOOL_OPS(netdev, &bnx2x_vf_ethtool_ops);
+ netdev->ethtool_ops = (IS_PF(bp)) ?
+  &bnx2x_ethtool_ops : &bnx2x_vf_ethtool_ops;
 }
-- 
1.7.1